packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -924,30 +987,79 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
   }
 
   Size _calculateChildSize(BoxConstraints constraints) {
+    final double childWidth = _getMaxChildWidth(constraints);
+    final double maxHeight = _getMaxChildHeight(constraints, childWidth);
+    return Size(childWidth, maxHeight);
+  }
+
+  double _getMaxChildWidth(BoxConstraints constraints) {
     final int childCount = this.childCount ~/ 2 + 1;
     double childWidth = (constraints.minWidth - totalSeparatorWidth) / childCount;
-    double maxHeight = _kMinSegmentedControlHeight;
     RenderBox? child = firstChild;
     while (child != null) {
       childWidth = math.max(childWidth, child.getMaxIntrinsicWidth(double.infinity) + 2 * _kSegmentMinPadding);
       child = nonSeparatorChildAfter(child);
     }
-    childWidth = math.min(
+    return math.min(
       childWidth,
       (constraints.maxWidth - totalSeparatorWidth) / childCount,
     );
-    child = firstChild;
+  }
+
+  double _getMaxChildHeight(BoxConstraints constraints, double childWidth) {
+    double maxHeight = _kMinSegmentedControlHeight;
+    RenderBox? child = firstChild;
     while (child != null) {
       final double boxHeight = child.getMaxIntrinsicHeight(childWidth);
       maxHeight = math.max(maxHeight, boxHeight);
       child = nonSeparatorChildAfter(child);
     }
-    return Size(childWidth, maxHeight);
+    return maxHeight;
   }
 
-  Size _computeOverallSizeFromChildSize(Size childSize, BoxConstraints constraints) {
+  List<double> _getChildIntrinsicWidths(BoxConstraints constraints) {
```

Comment:
nit: consider combining the call paths for the `!proportionalWidth` case into this method, so you only have to handle `proportionalWidth` once.

Reply:
Done! Thanks:)